src/psd_tools/api/layers.py

```diff
@@ -779,6 +757,38 @@ def count(self, layer):
 
         return self._layers.count(layer)
 
+    def _check_valid_layers(self, layers):
+        
+        assert layers is not self, "Cannot add the group {} to itself.".format(self)
+
+        if isinstance(layers, list):
+            for layer in layers:
+                assert isinstance(layer, Layer)
+                if layer.kind in ["group", "psdimage", "artboard"]: 
+                    assert self not in layer.descendants(), "This operation would create a reference loop within the group between {} and {}.".format(self, layer)
+
+        else:
+            assert isinstance(layers, Layer)
+            if layers.kind in ["group", "psdimage", "artboard"]: 
+                assert self not in layers.descendants(), "This operation would create a reference loop within the group between {} and {}.".format(self, value)
+
+    def _update_layer_metadata(self):
+
+        _psd = self if self.kind == "psdimage" else self._psd
+
+        for layer in self.descendants():
+            
+            if layer._psd != _psd and _psd is not None:
+                if layer.kind == "pixel":
+                    layer._convert(_psd)
+            
+                layer._fetch_patterns(_psd)
```

Comment:
While I understand the logic, I am not sure why this method only supports patterns. What is the plan for tagged block items? Is the approach here future-proof when considering other items?

Reply:
I only made the method for patterns because it's the only one encountered during a workflow, I have yet to find another global tagged block (because the patterns are stored in the layer_and_mask_info tagged blocks not in the layer's blocks) that would need to be copied from one psd to another and which conditions should it meet for it. The Adobe spec isn't very explicit on whether it uses a layer's tagged blocks or the global one to store data.

Now when it comes to future proofing, I can probably move part of the pattern fetching logic to the pattern or effect class other than that without more informations on what datas need to be moved I can't really tell

Reply:
Ok, perhaps it would be better to have a method like `_fetch_tagged_blocks` and handle patterns inside of it for the future

Reply:
I renamed the function into `_fetch_tagged_blocks`, i will leave its body the same for now and whenever a new type of data needed to be moved i'll refactor it to be more general.